conversations/user_interface.py

```diff
@@ -0,0 +1,228 @@
+            data=request.POST, files=request.FILES
+        )
+
+        if form.is_valid():
+            thread, _ = create_new_support_request(
+                conference=Conference.objects.current(),
+                requested_by=request.user,
+                title=form.cleaned_data["title"],
+                content=form.cleaned_data["content"],
+                # TODO: attachments
+            )
+            # TODO: messages.succeess
+            return redirect(thread.get_user_url())
+
+    return TemplateResponse(
+        request,
+        "ep19/bs/conversations/user_interface/start_new_thread.html",
+        {"new_support_request_form": form},
+    )
+
+
+@login_required
+def new_finaid_request(request):
+
+    form = UserNewFinaidRequest()
+
+    if request.method == "POST":
+        form = UserNewFinaidRequest(data=request.POST, files=request.FILES)
+
+        if form.is_valid():
+            thread, _ = create_new_finaid_request(
+                conference=Conference.objects.current(),
+                requested_by=request.user,
+                finaid_data=dict(**form.cleaned_data),
+            )
+            return redirect(thread.get_user_url())
+
+    return TemplateResponse(
+        request,
+        "ep19/bs/conversations/user_interface/new_finaid_request.html",
+        {"new_finaid_request_form": form},
+    )
+
+
+@login_required
+def user_thread(request, thread_uuid):
+    thread = Thread.objects.get(uuid=thread_uuid)
+
+    if thread.created_by != request.user:
+        raise Http404()
+
+    user_reply_form = UserReplyForm()
+
+    if request.method == "POST":
+
+        if ThreadActions.submit_reply_to_thread in request.POST:
+            user_reply_form = UserReplyForm(
+                data=request.POST, files=request.FILES
+            )
+
+            if user_reply_form.is_valid():
+                user_reply_to_thread(
+                    thread=thread,
+                    # replied_by=request.user,
+                    content=user_reply_form.cleaned_data["content"],
+                )
+                return redirect(".")
+
+    return TemplateResponse(
+        request,
+        "ep19/bs/conversations/user_interface/thread.html",
+        {
+            "thread": thread,
+            "ThreadActions": ThreadActions,
+            "user_reply_form": user_reply_form,
+        },
+    )
+
+
+urlpatterns = [
+    url(r"^threads/$", threads, name="threads"),
+    url(r"^threads/start-new/$", start_new_thread, name="start_new_thread"),
+    url(
+        r"^threads/start-new/finaid/$",
+        new_finaid_request,
+        name="new_finaid_request",
+    ),
+    url(r"^thread/(?P<thread_uuid>[\w-]+)/$", user_thread, name="user_thread"),
+]
+
+
+class UserReplyForm(forms.Form):
+    content = forms.CharField(widget=forms.Textarea)
+
+
+class UserCreateHelpdeskRequestForm(forms.Form):
+    title = forms.CharField()
+    content = forms.CharField(widget=forms.Textarea)
+
+
+FINAID_GRANT_CHOICES = Choices(
```

Comment:
If these module level variables are only used here, I guess they could be "private":

I mean just adding a `_` as convention for them.

Reply:
I'll put them under the form – they are used in tests (as enums) and possibly in some other places in the future.